code/__HELPERS/areas.dm

```diff
@@ -22,6 +22,8 @@ GLOBAL_LIST_INIT(typecache_powerfailure_safe_areas, typecacheof(/area/engine/eng
 		found_turfs.Cut(1, 2)
 		var/dir_flags = checked_turfs[sourceT]
 		for(var/dir in GLOB.alldirs)
+			if(length(.) > 2 * BP_MAX_ROOM_SIZE)
+				return
```

Comment:
This proc isnt just for blueprints, like I said last time this was suggested

Reply:
detect_room is only used by create_area as far as I can see though, so nothing else should be affected.

Reply:
You misunderstand, the proc itself has a simple purpose independent of the blueprints. This is useful for things other than it and one of the reasons I originaly wrote it was to seperate it out for some other features people wanted. 

Those didn't really turn out but the purpose of the proc is still there. Inserting snowflake exceptions like this into unrelated code is a perfect way to cultivate shitcode.

Reply:
Fair enough, I'll make a different fix for it. We do need the proc to stop early when called by create_area though, but I can add an argument that defaults to ignore size. Sound good to you?